components/NavBar.tsx

```diff
@@ -1,28 +1,37 @@
+"use client";
 import Image from "next/image";
 import Link from "next/link";
-import { Lobster } from "next/font/google";
-
-const lobster = Lobster({ weight: "400", subsets: ["latin"] });
-import Icon from "./Icon";
+import { useSession } from "next-auth/react";
+import { signIn, signOut } from "next-auth/react";
 
 export default function NavBar() {
+  const { status } = useSession();
   return (
     <header
-      className={`flex justify-between items-center border-b border-gray-800 p-4 ${lobster.className}`}
+      className={`flex justify-between items-center border-b border-gray-800 p-4`}
     >
       <Link href="/">
-        <h1 className="text-2xl">Collections</h1>
+        <h1 className="text-2xl font-lobster">Collections</h1>
       </Link>
 
-      <Link href="/user-profile">
-        <Image
-          src="/images/profile-picture.jpg"
-          width={32}
-          height={32}
-          alt="User profile picture"
-          className="rounded-full"
-        />
-      </Link>
+      {status === "authenticated" ? (
```

Comment:
if you're logged in and refresh the page, you see the "sign in" button briefly, then the logged-in state. It's because the sign-in button is being SSR'd.

I think a better way of doing this for now would be to not render anything for the sign-in / user state component until useSession is initialized